src/sp/protocol/mqtt/mqtt_parser.c

```diff
@@ -765,8 +765,9 @@ nmq_connack_encode(nng_msg *msg, conn_param *cparam, uint8_t reason)
 	nni_msg_append(msg, &reason, 1);
 
 	if (cparam->pro_ver == MQTT_PROTOCOL_VERSION_v5) {
-		// TODO set properties if necessary. Null by default.
-		encode_properties(msg, NULL, CMD_CONNACK);
+		// TODO change properties if necessary.
+		// modify it in nego_cp while init cparam
+		encode_properties(msg, cparam->properties, CMD_CONNACK);
```

Comment:
Not allo properties in connect are allowed in connack

Reply:
Don't be lazy ^_^

Reply:
oh shit